src/charm.py

```diff
@@ -532,15 +564,43 @@ def _on_set_password(self, event: ActionEvent) -> None:
                 event.fail(f"Failed changing the password: {e}")
                 return
 
-        self.set_secret("app", MongoDBUser.get_password_key_name_for_user(username), new_password)
+        secret_id = self.set_secret(
+            APP_SCOPE, MongoDBUser.get_password_key_name_for_user(username), new_password
+        )
 
         if username == BackupUser.get_username():
             self._connect_pbm_agent()
 
         if username == MonitorUser.get_username():
             self._connect_mongodb_exporter()
 
-        event.set_results({Config.Actions.PASSWORD_PARAM_NAME: new_password})
+        event.set_results(
+            {Config.Actions.PASSWORD_PARAM_NAME: new_password, "secret-id": secret_id}
```

Comment:
Semi-nitpick: we are actually using the secret URI. I'm wondering if we may wanna change it for the event contract, for clarity? [It's not a very important, rather almost a hypotetical question :-) ]